adapter/src/dbt/adapters/fal_experimental/adapter.py

```diff
@@ -47,16 +47,25 @@ def run_in_environment_with_adapter(
     manifest: Manifest,
     macro_manifest: MacroManifest
 ) -> AdapterResponse:
-    from isolate.backends.remote import IsolateServer
     """Run the 'main' function inside the given code on the
     specified environment.
 
     The environment_name must be defined inside fal_project.yml file
     in your project's root directory."""
-    if type(environment) == IsolateServer:
-        # TODO: make a specialized function for this case?
+
+    if type(environment).__name__ in ['IsolateServer', 'FalHostedServer']:
         deps = [i for i in get_default_pip_dependencies() if i.startswith('dbt-')]
 
+        if not any('dbt-fal' in i for i in deps):
```

Comment:
in `get_default_pip_dependencies` there is already a special treatment of `dbt-fal`. Can we solve the hack in there instead of bringing it here?

Reply:
Yeah, I thought about this, but then I realized that installing dbt-fal is different depending on where isolate is running. This case is especially special because it would happen only if dbt-fal is not installed from pip, but rather locally. And it's OK if isolate is also local, we can just point it to the local dbt-fal directory (as we do in get_default_pip_dependencies), but here isolate is remote. So we either have to upload the local version of dbt-fal or we just install whatever is available from pip. I think that we should definitely print a warning here though. Maybe this is a special enough case to live here?